sentry/src/main/java/io/sentry/adapters/TransactionContextsDeserializerAdapter.java

```diff
@@ -0,0 +1,122 @@
+package io.sentry.adapters;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.SpanContext;
+import io.sentry.TransactionContext;
+import io.sentry.TransactionContexts;
+import io.sentry.protocol.App;
+import io.sentry.protocol.Browser;
+import io.sentry.protocol.Device;
+import io.sentry.protocol.Gpu;
+import io.sentry.protocol.OperatingSystem;
+import io.sentry.protocol.SentryRuntime;
+import java.lang.reflect.Type;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class TransactionContextsDeserializerAdapter
+    implements JsonDeserializer<TransactionContexts> {
+
+  private final @NotNull ILogger logger;
+
+  public TransactionContextsDeserializerAdapter(@NotNull final ILogger logger) {
+    this.logger = logger;
+  }
+
+  @Override
+  public TransactionContexts deserialize(
+      JsonElement json, Type typeOfT, JsonDeserializationContext context)
+      throws JsonParseException {
+    try {
+      if (json != null && !json.isJsonNull()) {
```

Comment:
should we extract the content of this method to a Util class and reuse it in `ContextsDeserializerAdapter` and `ContextsSerializerAdapter`? they look pretty much the same.

Reply:
I'll see what i can do - they don't share the same class hierarchy, at least not at the moment.